code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm

```diff
@@ -0,0 +1,251 @@
+
+/**
+  * #Eldritch Knwoledge
+  *
+  * Datum that makes eldritch cultist interesting.
+  *
+  * Eldritch knowledge aren't instantiated anywhere roundstart, and are initalized and destroyed as the round goes on.
+  */
+/datum/eldritch_knowledge
+	///Name of the knowledge
+	var/name = "Basic knowledge"
+	///Description of the knowledge
+	var/desc = "Basic knowledge of forbidden arts."
+	///What shows up
+	var/gain_text = ""
+	///Cost of knowledge in souls
+	var/cost = 0
+	///Next knowledge in the research tree
+	var/list/next_knowledge = list()
+	///What knowledge is incompatible with this. This will simply make it impossible to research knowledges that are in banned_knowledge once this gets researched.
+	var/list/banned_knowledge = list()
+	///Used with rituals, how many items this needs
+	var/list/required_atoms = list()
+	///What do we get out of this
+	var/list/result_atoms = list()
+	///What path is this on defaults to "Side"
+	var/route = "Side"
+
+/datum/eldritch_knowledge/New()
+	. = ..()
+	var/list/temp_list
+	for(var/X in required_atoms)
+		var/atom/A = X
+		temp_list += list(typesof(A))
```

Comment:
This is pretty pointless, if you don't need any subtype blacklist format you can just do istype in ritual part.

Reply:
im doing this because the ritual also accepts subtypes. The way you make new ritual is you add a type to the required_atoms list, and ALL subtypes of that atom will be accepted in the ritual.